doc/tutorials/raster-reprojection.md

```diff
@@ -0,0 +1,131 @@
+
+# Advanced
+
+### Disabling reprojection
+In case you are creating a custom build of OpenLayers and do not need the reprojection code, you can reduce the build size by setting `ol.ENABLE_RASTER_REPROJECTION` to `false`, which completely disables the reprojection support.
+See [Custom builds](custom-builds.html#defines) tutorial to see how to do this.
+
+### Triangulation precision threshold
+The default [triangulation error threshold](#dynamic-triangulation) in pixels is given by `ol.DEFAULT_RASTER_REPROJECTION_ERROR_THRESHOLD` (0.5 pixel).
+In case different threshold needs to be defined for different sources, `reprojectionErrorThreshold` option can be passed when constructing tile image source.
+
+###Limiting visibility of reprojected map by extent
+
+The reprojection algorithm uses inverse transformation (from *view projection* to *data projection*).
+For certain coordinate systems this can result in a "double occurrence" of the source data on a map.
+For example, when reprojecting a map of Switzerland from EPSG:21781 to EPSG:3857, it is displayed twice: once at the proper place in Europe, but also in the Pacific Ocean near the New Zealand, on the opposite side of the globe.
+
+<img src="raster-reprojection-resources/double-occurrence.jpg" alt="Double occurrence of a reprojected map" width="600" />
+
+Although this is mathematically correct behavior of the inverse transformation, visibility of the map on multiple places is not expected by users.
+A possible general solution would be to calculate the forward transformation for every vertex as well - but this would significantly decrease performance (especially for computationally expensive transformations).
+
+Therefore a recommended workaround is to define a proper visibility extent on the `ol.layer.Tile` in the view projection.
+Setting of such limit is demonstrated in the [reprojection demo example](http://openlayers.org/en/master/examples/reprojection.html).
+
+### Resolution calculation
+When determining source tiles to load, the ideal source resolution needs to be calculated.
+The `ol.reproj.calculateSourceResolution(sourceProj, targetProj, targetCenter, targetResolution)` function calculates the ideal value in order to achieve pixel mapping as close as possible to 1:1 during reprojection, which is then used to select proper zoom level from the source.
+
+It is, however, generally not practical to use the same source zoom level for the whole target zoom level -- different projections can have significantly different resolutions in different parts of the world (e.g. polar regions in EPSG:3857 vs EPSG:4326) and enforcing single resolution for whole zoom level would result in some tiles being scaled up/down, possibly requiring huge number of source tiles to be loaded.
+Therefore, the resolution mapping is calculated for each reprojected tile separately (in the middle of the tile extent).
```

Comment:
calculated for each reprojected tile separately -> calculated separately for each reprojected tile